src/platform_impl/linux/x11/ime/callbacks.rs

```diff
@@ -136,13 +137,17 @@ pub unsafe extern "C" fn xim_instantiate_callback(
     let inner: *mut ImeInner = client_data as _;
     if !inner.is_null() {
         let xconn = &(*inner).xconn;
-        let result = replace_im(inner);
-        if result.is_ok() {
-            let _ = unset_instantiate_callback(xconn, client_data);
-            (*inner).is_fallback = false;
-        } else if result.is_err() && (*inner).is_destroyed {
-            // We have no usable input methods!
-            result.expect("Failed to reopen input method");
+        match replace_im(inner) {
+            Ok(()) => {
+                let _ = unset_instantiate_callback(xconn, client_data);
+                (*inner).is_fallback = false;
+            }
+            Err(err) => {
+                if (*inner).is_destroyed {
+                    // We have no usable input methods!
+                    panic!("Failed to reopen input method: {:?}", err);
```

Comment:
Debug formatting is probably the correct "direct" translation, but is it the correct formatting option here?

Reply:
yes, I think so.
Not sure what else I could do here.

Reply:
Well, the alternative is "normal" formatting, but it's not that big of a deal either way.

Reply:
I would just try it if I were at a computer but I'm pretty sure you can't just normal format an enum unless it implements display

Reply:
Ah, alright, that's my bad for not checking what the error type is.